src/zope/app/apidoc/codemodule/browser/README.rst

```diff
@@ -44,7 +44,7 @@ Return info objects for all classes in this module.
   >>> pprint(details.getClasses())
   [{'doc': 'Represent the code browser documentation root',
     'name': 'CodeModule',
-    'url': 'http://127.0.0.1/++apidoc++/Code/zope/app/apidoc/codemodule/codemodule/CodeModule'}]
+    'url': 'http://127.0.0.1/++etc++site/++apidoc++/Code/zope/app/apidoc/codemodule/codemodule/CodeModule'}]
```

Comment:
The latter. It directly calls `zope.location.location.locate()` on the child every time one is looked up.

Using a LocationProxy doesn't work because the children themselves *also* dynamically create children and set their `__parent__`, and so the chain is broken if the children don't have their own correct `__parent__` (only the proxy object does).